src/core/reference/src/runtime/reference/irdft.cpp

```diff
@@ -0,0 +1,166 @@
+    const auto outer_strides = remove_from_position(input_data_strides, reversed_last_axis);
+
+    const auto outer_extended_strides = remove_from_position(reversed_ext_input_strides, reversed_last_axis);
+    const auto inner_stride = input_data_strides[reversed_last_axis];
+    const auto inner_extended_stride = reversed_ext_input_strides[reversed_last_axis];
+
+    const auto inner_bound = last_signal_size / 2 + 1;
+    const auto inner_size = reversed_input_data_shape[reversed_last_axis];
+    const auto inner_extended_size = reversed_ext_input_data_shape[reversed_last_axis];
+
+    std::vector<complex_type> extended_input_complex(shape_size(shape_of_extended_input_data) / 2, complex_type{0, 0});
+    for (int64_t i = 0; i < outer_extended_size; ++i) {
+        const auto outer_coords = fft_common::coords_from_index(i, outer_extended_shape_strides);
+        const auto outer_input_offset = fft_common::offset_from_coords_and_strides(outer_coords, outer_strides);
+        const auto outer_output_offset =
+            fft_common::offset_from_coords_and_strides(outer_coords, outer_extended_strides);
+
+        for (int64_t j = 0; j < inner_bound; ++j) {
+            complex_type value = complex_type(0.0, 0.0);
+            if (j < inner_size) {
+                value = complex_input_ptr[outer_input_offset + j * inner_stride];
+            }
+            extended_input_complex[outer_output_offset + j * inner_extended_stride] = value;
+            if (j > 0 && j < (inner_extended_size - inner_bound + 1)) {
+                extended_input_complex[outer_output_offset + (inner_extended_size - j) * inner_extended_stride] =
+                    std::conj(value);
+            }
+        }
+    }
+    return extended_input_complex;
+}
+
+}  // namespace
+
+void irdft(const std::vector<float>& input_data,
+           const Shape& input_data_shape,
+           const std::vector<int64_t>& axes_data,
+           float* irdft_result,
+           const Shape& fft_output_shape,
+           int64_t last_signal_size) {
+    // calculate inverse FFT over the outer axes
+    const auto outer_ifft_axes = get_outer_fft_axes(axes_data);
+    auto outer_ifft_shape = input_data_shape;
+    for (const auto& a : outer_ifft_axes) {
+        outer_ifft_shape[a] = fft_output_shape[a];
+    }
+    std::vector<float> outer_fft_result = std::vector<float>(shape_size(outer_ifft_shape), 0.0f);
+    fft(reinterpret_cast<const float*>(input_data.data()),
+        input_data_shape,
+        outer_ifft_axes.data(),
+        Shape{outer_ifft_axes.size()},
+        outer_fft_result.data(),
+        outer_ifft_shape,
+        FFTKind::Inverse);
+
+    // adjust the input to Hermitian-symmetric size
+    const auto last_axis = axes_data.back();
+    auto extended_data_shape = outer_ifft_shape;
+    extended_data_shape[last_axis] = last_signal_size;
+    std::vector<complex_type> inner_ifft_result(shape_size(fft_output_shape) / 2);
```

Comment:
Yes, it is always even, because the last dim is 2 (representation of real and imaginary part), but good point - to make it more clear, now it is `shape_size(irdft_output_shape)`